cli/kv.go

```diff
@@ -143,7 +143,43 @@ func runPut(cmd *cobra.Command, args []string) {
 	}
 }
 
-// A incCmd command increments the value for one or more keys.
+// A cPutCmd command conditionally sets a value for a key.
+var cPutCmd = &cobra.Command{
+	Use:   "cput [options] <key> <value> [<expValue>]",
+	Short: "conditionally sets a value for a key",
+	Long: `
+Conditionally sets a value for a key if the existing value is equal
+to expValue. To conditionally set a value only if there is no existing entry
+pass nil for expValue. The expValue defaults to 1 if not specified.
+`,
+	Run: runCPut,
+}
+
+func runCPut(cmd *cobra.Command, args []string) {
+	if len(args) != 2 && len(args) != 3 {
+		cmd.Usage()
+		return
+	}
+
+	key := proto.Key(unquoteArg(args[0], true /* disallow system keys */))
+	value := proto.Key(unquoteArg(args[1], false))
+	var expValue proto.Key
```

Comment:
I ended up not pulling this PR into #2160 because of this question; take your time addressing it